### PR TITLE
Set up midstream mechanics for tracking upstream repo

### PR DESCRIPTION
**Pull request by JasonPowr**
This pr is related to this issue [here](https://github.com/securesign/image-factory/issues/18), which involves configuring mirroring in this repo similar to other repositories in the secure sign org.

## Comments 
When testing on my fork I was able to generate the `release-next` and `release-next-ci` branches successfully

### CODE REVIEW

File: redhat/overlays/cosign/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#Build stage
+FROM golang:1.20 AS build-env
```
Comment (lance) [inline redhat/overlays/cosign/Dockerfile:2]: This should use a Red Hat image. E.g. https://github.com/securesign/trillian/blob/fd0b9bb3dec783f1c2012fb036364d5ca68b48c3/redhat/overlays/log_server/Dockerfile#L1

**Review by lance — CHANGES_REQUESTED** [review]
Looks good - thanks! Just one small change to the base image reference, please.

File: redhat/overlays/cosign/Dockerfile (hunk 1 above)
Comment (JasonPowr) [reply redhat/overlays/cosign/Dockerfile:2]: No problem, I've also noticed that you have made changes in the securesign/rekor repo to how the branching is set up should I go ahead and incorporate those changes here as well as in the image factory ?

Comment (lance) [reply redhat/overlays/cosign/Dockerfile:2]: Yes - I would use the `redhat-v2.1.1` branch in image-factory to pin this to the latest release.

**Review by lance — APPROVED** [review]
/lgtm

**Pull request merged**